src/py_interval.cpp

```diff
@@ -10,10 +10,12 @@ Copyright (C) 2016-2023 Deep Genomics Inc. All Rights Reserved.
 #define GK_CHECK_REFG(a, b) \
 	GK_CHECK((a).refg == (b).refg, value, "Coordinate system mismatch, {} and {}.", (a), (b));
 
-#define GKPY_INTERVAL_BOOL_METHOD_ONEARG(pyname, method) \
-	GKPY_METHOD_BEGIN_ONEARG(pyname, method) \
-		const Py##pyname::value_t& c = Py##pyname::value(selfo); \
-		if (PyInterval::check(arg)) { const interval_t& a = PyInterval::value(arg); GK_CHECK_REFG(c, a); GKPY_RETURN_BOOL(c.method(a)); } \
+#define GKPY_INTERVAL_BOOL_METHOD_ONEARG(method) \
+	GKPY_METHOD_BEGIN_ONEARG(Interval, method) \
+		const PyInterval::value_t& x = PyInterval::value(selfo); \
+		auto y = PyInterval::value(arg); \
+		GK_CHECK(y.same_chrom(x) && y.same_strand(x), value, "Cannot call {} on intervals {} and {}: different strands or coordinate systems.", #method, x, y); \
```

Comment:
I would just return None rather than throwing an error personally.

Also I'm not sure different strands should have no intersection, but it's ambiguous which strand you would return the result on.

Reply:
Actually I see that test_interval.py test_intersect() [explicitly tests](https://github.com/deepgenomics/GenomeKit/blob/2f85841c39c48a206efb18d9dd7eec67ce522047/tests/test_interval.py#L695-L694) different strands and different *genomes* and returns None.
I limited the scope and changed to return None rather than raising, and added a test where needed.